src/__tests__/backend/transliterator/transliterator.lambda.test.ts

```diff
@@ -27,260 +23,135 @@ afterEach((done) => {
+  forPackage.mockImplementation(async (target: string) => {
+    return new MockDocumentation(target) as unknown as Documentation;
   });
 
-  // WHEN
-  const result = handler(payload, mockContext);
-
-  // THEN
-  await expect(result).resolves.toEqual([{ bucket: payload.Records[0].s3.bucket.name, key, versionId: 'New-VersionID' }]);
-
-  expect(mockSpawn).toHaveBeenCalled();
-  expect(mockTransliterateAssembly).toHaveBeenCalled();
-});
-
-test('unscoped package', async () => {
   // GIVEN
   const packageName = 'package-name';
   const packageVersion = '1.2.3-dev.4';
-  const payload: S3Event = {
+  const event: S3Event = {
     Records: [{
       awsRegion: 'bemuda-triangle-1',
-      eventVersion: '1337',
-      eventSource: 's3:DummySource',
-      eventName: 's3:DummyEvent',
-      eventTime: '1789-07-14T00:00:00+02:00',
-      userIdentity: { principalId: 'aws::principal::id' },
-      requestParameters: { sourceIPAddress: '127.0.0.1' },
-      responseElements: {
-        'x-amz-id-2': '456',
-        'x-amz-request-id': '123',
-      },
       s3: {
         bucket: {
           name: 'dummy-bucket',
-          arn: 'arn:aws:s3:::dummy-bucket',
-          ownerIdentity: { principalId: 'aws::principal::id' },
         },
-        configurationId: '42',
         object: {
-          eTag: 'eTag',
-          key: `${constants.STORAGE_KEY_PREFIX}${packageName}/v${packageVersion}${constants.PACKAGE_KEY_SUFFIX}`,
-          sequencer: 'Seq',
-          size: 1337,
+          key: `${constants.STORAGE_KEY_PREFIX}${packageName}/v${packageVersion}${constants.ASSEMBLY_KEY_SUFFIX}`,
           versionId: 'VersionId',
         },
-        s3SchemaVersion: '1',
       },
     }],
-  };
-  const mockContext = {} as any;
-  const mockTarballBytes = randomBytes(128);
-  const mockOutputAssembly = randomBytes(128);
+  } as any;
+
+  const assembly: spec.Assembly = {
+    targets: { python: {} },
+    submodules: { '@scope/package-name.sub1': {}, '@scope/package-name.sub2': {} },
+  } as any;
+
+  // mock the assembly request
+  mockFetchAssembly(assembly);
+
+  // mock the file uploads
+  mockPutDocs(
+    '/docs-python.md',
+    '/docs-sub1-python.md',
+    '/docs-sub2-python.md',
+    '/docs-typescript.md',
+    '/docs-sub1-typescript.md',
+    '/docs-sub2-typescript.md',
+  );
+
+  const created = await handler(event, {} as any);
+  expect(created.length).toEqual(6);
+  expect(created[0].key).toEqual(`data/${packageName}/v${packageVersion}/docs-python.md`);
+  expect(created[1].key).toEqual(`data/${packageName}/v${packageVersion}/docs-sub1-python.md`);
+  expect(created[2].key).toEqual(`data/${packageName}/v${packageVersion}/docs-sub2-python.md`);
+  expect(created[3].key).toEqual(`data/${packageName}/v${packageVersion}/docs-typescript.md`);
+  expect(created[4].key).toEqual(`data/${packageName}/v${packageVersion}/docs-sub1-typescript.md`);
+  expect(created[5].key).toEqual(`data/${packageName}/v${packageVersion}/docs-sub2-typescript.md`);
```

Comment:
Maybe would read nicer as §this also renders the length check redundant):

```suggestion
  expect(created.map(({ key }) => key).toEqual([
    `data/${packageName}/v${packageVersion}/docs-python.md`,
    `data/${packageName}/v${packageVersion}/docs-sub1-python.md`,
    `data/${packageName}/v${packageVersion}/docs-sub2-python.md`,
    `data/${packageName}/v${packageVersion}/docs-typescript.md`,
    `data/${packageName}/v${packageVersion}/docs-sub1-typescript.md`,
    `data/${packageName}/v${packageVersion}/docs-sub2-typescript.md`,
  ]);
```

Reply:
Nice